src/Runtime/LambdaRuntime.php

```diff
@@ -43,38 +43,23 @@ final class LambdaRuntime
     /** @var Invoker */
     private $invoker;
 
-    /** @var int seconds */
-    private $timeout;
+    /** @var bool */
+    private $enableTimeout;
 
-    public static function fromEnvironmentVariable(?int $timeout = null): self
+    public static function fromEnvironmentVariable(): self
     {
-        if ($timeout === null) {
-            $envTimeout = getenv('BREF_TIMEOUT');
-            if ($envTimeout === false || $envTimeout === '') {
-                // In 1.3 the Timeout exception is opt-in only
-                $timeout = -1;
-            } else {
-                $timeout = (int) $envTimeout;
-            }
-        }
-
-        return new self((string) getenv('AWS_LAMBDA_RUNTIME_API'), $timeout);
+        return new self((string) getenv('AWS_LAMBDA_RUNTIME_API'));
     }
 
-    /**
-     * @param int $timeout number of seconds before a TimeoutException is thrown.
-     *                     Value -1 means "disabled". Value 0 means "auto", this will
-     *                     set the timeout just a bit shorter than the Lambda timeout.
-     */
-    public function __construct(string $apiUrl, int $timeout = 0)
+    public function __construct(string $apiUrl)
     {
         if ($apiUrl === '') {
             die('At the moment lambdas can only be executed in an Lambda environment');
         }
 
         $this->apiUrl = $apiUrl;
         $this->invoker = new Invoker;
-        $this->timeout = $timeout;
+        $this->enableTimeout = (getenv('BREF_FEATURE_TIMEOUT') !== false);
```

Comment:
I strongly think we should be able to construct a `LambdaRuntime` without using environment variables. That will make testing easier, allow us to be thread safe and give the creator more control. 

I suggest moving this to a named factory method.

Reply:
Right, I see. TBH my goal is to have simple logic. If we have an internal boolean flag to control the timeout, that means we have 2 different options for the same thing:

- the boolean flag
- the boolean `BREF_FEATURE_TIMEOUT` env variable

That makes 4 possible combinations, instead of 2. Not much, but 2 possibilities are simpler to deal with than 4.

What would we gain if we added this little extra complexity? I don't think testing would be impacted, given tests already exist and work. I'm also happy if I can write tests as high level as possible (using the same API as users when possible).

Regarding thread safety, I don't see where multithreading the lambda runtime class would be used in the wild. We could also solve that by using `$_SERVER` if that ever happens.

So to me it boils down to: there's a single feature flag for the feature instead of 2.

Reply:
I understand that you would like to have simplicity.

I still think it is a mistake to let a class' internal behaviour to be affected by the global state. I dont follow your logic about the increased combinations. What I suggest is the following: 

```
 public static function fromEnvironmentVariable(): self
    {
        return new self((string) getenv('AWS_LAMBDA_RUNTIME_API'), getenv('BREF_FEATURE_TIMEOUT') !== false);
    }

    public function __construct(string $apiUrl, bool $enableTimeout /* maybe a default value */)
    {
        if ($apiUrl === '') {
            die('At the moment lambdas can only be executed in an Lambda environment');
        }

        $this->apiUrl = $apiUrl;
        $this->invoker = new Invoker;
        $this->enableTimeout = $enableTimeout;
    }
```

Reply:
OK I see, with the example indeed it's more concise that what I pictured.

I tried to go with it, but that got me here:

<img width="937" alt="image" src="https://user-images.githubusercontent.com/720328/112762816-16b18880-9002-11eb-9527-d6282a4aa276.png">

On one hand, that's cool because I can get rid of changing the global state for the tests, and pass `true/false` explicitly.

But on the other hand, what I want to test is that the `BREF_FEATURE_TIMEOUT` setting actually works. So I want to do a very high level test, I consider it definitely ok here to change the environment variable because I would be testing the same API that end users actually use.

FTR I was going to say there's no downside, but the last commit I just pushed will show that it's not true: I have to be careful resetting the global state. But it still feels more comfortable with a higher level test here.